source/options.tsx

```diff
@@ -17,93 +16,14 @@ import {perDomainOptions} from './options-storage.js';
 import isDevelopmentVersion from './helpers/is-development-version.js';
 import {doesBrowserActionOpenOptions} from './helpers/feature-utils.js';
 import {state as bisectState} from './helpers/bisect.js';
-import {parseTokenScopes} from './github-helpers/github-token.js';
 import {scrollIntoViewIfNeeded} from './github-helpers/index.js';
 import initFeatureList, {updateListDom} from './options/feature-list.js';
-
-type TokenType = 'classic' | 'fine_grained';
+import initTokenValidation from './options/token-validation.js';
 
 let syncedForm: SyncedForm | undefined;
 
-type Status = {
-	tokenType: TokenType;
-	error?: true;
-	text?: string;
-	scopes?: string[];
-};
-
 const {version} = chrome.runtime.getManifest();
 
-function reportStatus({tokenType, error, text, scopes}: Status): void {
```

Comment:
Extract all code related to reporting to its own file, to help https://github.com/refined-github/refined-github/issues/6446 and to facilitate reuse in other extensions. I'll try to create a `<github-token>` web component if not too difficult.